runtime/src/accounts_db.rs

```diff
@@ -3688,7 +3720,7 @@ impl AccountsDB {
                     "AccountDB::accounts_index corrupted. Storage pointed to: {}, expected: {}, should only point to one slot",
                     store.slot(), *slot
                 );
-                let count = store.remove_account(account_info.stored_size);
+                let count = store.remove_account(account_info.stored_size, reset_accounts);
```

Comment:
@sakridge well, I pondered some time for the possibility where dead slot's stores are recycled too prematurely. If we could allow that, it's still possible to `.reset()` when using via background-hashing. In conclusion, it seems that this isn't possible, right? how about mentioning at one of new comments in some way?

Well, AccountsDB is quite complex. I wonder now that we have AccountsCaching, recycling code could be a worth thing to maintain...?

Reply:
Hhh, I rather now think the bad recycle could happen in fact? Could you write a simple test for this?

Reply:
Yes, the AccountStorageEntry can be recycled while the hashing is in-progress, but it will not be taken out because the Arc ref will be >1 as from this check:

https://github.com/solana-labs/solana/blob/971c222cf7618559761bbdad63256a803171da85/runtime/src/accounts_db.rs#L2153

The AccountsPackage and the recycler will both hold a reference to the store.

Reply:
oh, thanks for the explanation. makes sense.